### PR TITLE
Create pipeline for CVP extension

### LINKED ISSUE #348: Develop CD Pipeline for CVP Application
**Is your feature request related to a problem? Please describe.**
Create CD pipeline for CVP application.

**Describe the solution you'd like**
Each time a Vane release is created CD pipeline will start and create a CVP application.  Application will be installable using cvpi on CVP.

**Describe alternatives you've considered**
None

**Additional context**
None

### PR DESCRIPTION
**Pull request by grybak-arista**
# Please include a summary of the changes

* .github/workflows/cvp-extension-builder.yml - pipeline that builds the RPM package for CVP
* Dockerfile - Vane development container (reworked to operate like the CVP container when running)
* Dockerfile.CVP - container that will run in the CVP environment
* Dockerfile.build-CVP - build environment container in which RPM package is created
* Makefile - updated with make targets for building CVP rpm, and cleaned up some existing make targets
* conf/cvpi/conf (directory) - added required files for CVP extension
* resources/vane-bashrc - created a bashrc file that is copied to Dockerfile(.CVP) containers to customize the environment
* vane-cvp_cvpinstall.spec - added RPM spec file for building CVP RPM

# Any specific logic/part of code you need extra attention on

Review the logic of the pipeline file in .github/workflows and the Dockerfiles to make sure that the process makes sense.

# Include the Issue number and link

Closes #348 

# List/Attach any dependencies/past issues that are required for this change/provide context

N/A

# Type of change

- - [ ] Bug fix (non-breaking change which fixes an issue)
- - [X] New feature (non-breaking change which adds functionality)
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- - [ ] This change requires a documentation update
- - [ ] Other (please specify)

# Effort required on reviewers end

- - [ ] Easy
- - [X] Medium
- - [ ] Hard 

# How Has This Been Tested?

## New feature

Pipeline tested in non-default branch using push trigger. Resulting RPM downloaded and installed on a CVP host running in ACT and verified. Screen grabs from installation and login to container below.

    
Screenshot 2023-08-08 at 12 49 08 PM

Screenshot 2023-08-08 at 12 49 50 PM

Screenshot 2023-08-08 at 12 50 17 PM

Screenshot 2023-08-08 at 12 50 42 PM

# CI pipeline result

- - [X] Pass
- - [ ] Fail
  
  If it fails, explain the reason and whether or not we should ignore the failure
  
# Verify Documentation Update

    If applicable, ensure documentation has been updated for ReadMe, Getting Started guide, and Style guide
    
# Additional comments

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -13,6 +15,12 @@ ENV PIP_NO_CACHE_DIR 1
 # Set the poetry version to use
 ENV POETRY_VERSION 1.5.1
```
Comment (rewati-arista) [inline Dockerfile:16]: We should lock down the POETRY version we plan on using, I was currently on 1.4.2 which is the one I included in the installation script: https://github.com/aristanetworks/vane/blob/68d8df7b8955381a3293a3695b8709d64f69b6d7/install_vane.sh#L112
We can discuss in the dev sync today

File: Dockerfile.build-CVP [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+# Start with a Centos 7 distro (what CVP is deployed on)
+FROM centos:7
+
+# Force a failure in any stage of a piped command to trigger a failure
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+
+# Set the prompt
+ENV PS1='cvp-rpm-build # '
+
+# Disable the mirrorlist because they are useless
```
Comment (rewati-arista) [inline Dockerfile.build-CVP:14]: might want to rephrase this comment hahah

**Review by rewati-arista — APPROVED** [review]
LGTM

**Review by shachiagarwal — APPROVED** [review]
LGTM

File: Dockerfile.build-CVP (hunk 2 above)
Comment (grybak-arista) [reply Dockerfile.build-CVP:14]: Leaving this for now (it was part of the original dockefile I used as a starting point)

**Pull request merged**